gpt_index/indices/query/base.py

```diff
@@ -248,7 +248,10 @@ def retrieve(self, query_bundle: QueryBundle) -> List[NodeWithScore]:
         similarity_tracker = SimilarityTracker()
         nodes = self._retrieve(query_bundle, similarity_tracker=similarity_tracker)
 
-        postprocess_info = {"similarity_tracker": similarity_tracker}
+        postprocess_info = {
+            "similarity_tracker": similarity_tracker,
+            "query_bundle": query_bundle,
+        }
```

Comment:
I would advise against passing arguments via unstructured dict.

This makes the interface leaky, and hard to reason.

Reply:
fair. would you recommend just passing the query string as an argument then?